aten/src/ATen/native/mps/OperationUtils.h

```diff
@@ -197,13 +199,39 @@ struct MPSGraphCache
     });
     return result;
   }
+
+  bool FindAndRemoveViewEntry(void* ptr) {
+    bool removed_entry = false;
+
+    // this may find multiple view entries with the same buffer pointers
+    auto views_range = views_list.equal_range(ptr);
+    if (views_range.first != views_range.second) {
+      for (auto view_it = views_range.first; view_it != views_range.second; ++view_it) {
+        MPSCacheKey hash = view_it->second;
+        // find the cache entry associated with the hash
+        auto cache_it = cache_.find(hash);
+        if (cache_it != cache_.end()) {
+          cache_.erase(cache_it);
+          delete cache_it->second.cachedGraph_;
+          removed_entry = true;
+        }
+      }
+      // this erase-by-key will remove all pairs in the list with the same key
+      views_list.erase(ptr);
+    }
+    return removed_entry;
+  }
+
  private:
   MPSGraphCache() {
     serialQueue_ = dispatch_queue_create("cache queue", DISPATCH_QUEUE_SERIAL);
   }
 
   static MPSGraphCache* _instance_cache;
   std::unordered_map<MPSCacheKey, CacheEntry> cache_;
+  // list of buffers associated with view entries in the cache
+  // note that multiple view cache entries could use the same buffer pointer
+  std::multimap<void*, MPSCacheKey> views_list;
```

Comment:
Here is there a reason to use void*? I am assuming this is storing the data() address here, so can just use:
```
<uintptr_t, MPSCacheKey>
```
It will be easier  for the stdlib to efficiently store and do key compare on multimap.  
The other thing is why do we need to use multi map here ? Can't we just use flat hash map or unordered_map?
std::unordered_map<uintptr_t, std::vector<MPSCacheKey>> to store all the keys. We are going to just go and delete them from the cache right.

Reply:
I changed multimap to unordered_multimap.